ClosedXML/Excel/DataValidation/XLDataValidations.cs

```diff
@@ -44,18 +130,34 @@ public Boolean ContainsSingle(IXLRange range)
             return count == 1;
         }
 
-        #endregion IXLDataValidations Members
-
         public void Delete(IXLDataValidation dataValidation)
         {
-            _dataValidations.RemoveAll(dv => dv.Ranges.Equals(dataValidation.Ranges));
+            if (!_dataValidations.Remove(dataValidation))
+                return;
+            var xlDataValidation = dataValidation as XLDataValidation;
+            xlDataValidation.RangeAdded -= OnRangeAdded;
```

Comment:
Should this class not implement `IDisposable` and events be unhooked upon disposal? Remind me what's best practices for event handlers, please.

Reply:
Agree. Events ought to be unsubscribed. It may make no difference in 99% of cases but the rest 1% may be a real pain in the neck, so it is safer to always unsubscribe explicitly.
In this particular case, if I am not mistaking, preserving a single reference to the data validation rule may prevent the whole collection from being garbage collected. 
WIll fix.

Reply:
After reconsideration, I think we don't need `IDisposable` here: the user could dispose of DV persisting in the collection, thus putting it into an invalid state (it would not reflect changes in ranges on editing or shifting).
I found, however, another bug related to subscription/unsubscription during the consolidation. Now it is fixed.

Also, I have rebased this branch.